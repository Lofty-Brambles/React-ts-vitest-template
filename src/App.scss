@use "styles/mix";
@use "styles/var";

.App {
	@include mix.flex(column);
	padding: 3rem;
	text-align: center;

	& .react {
		@include mix.box(10rem);
		animation: spin infinite 20s linear;
	}

	& h1 {
		font-family: var.$titan;
		font-size: 2rem;
	}

	& span {
		white-space: nowrap;
	}

	& ul {
		list-style-type: none;
		padding: 0;

		& > li {
			padding: 0.25rem;
			counter-increment: item;
			font-family: var.$flower;
			font-weight: 600;

			&::before {
				content: counter(item);
				display: inline-block;
				width: 12px;
				height: 20px;
				font-family: "Indie Flower", cursive;
				margin-right: 5px;
				background-color: #f9dd94;
				color: #7eb4e2;
				font-weight: bold;
				font-size: 140%;
				padding: 0 8px 8px;
				border-radius: 3px;
				line-height: 1.3;
			}
		}
	}

	& .card {
		& button {
			padding: 0.75rem 1.25rem;
			outline: 0;
			border: 1px solid var.$black;
			position: relative;

			transform: rotate(-2deg);

			background-color: var.$cyan;
			font-family: var.$fedora;
			font-weight: 800;
			color: var.$white;
			user-select: none;
			touch-action: manipulation;

			&::after {
				content: "";
				border: 1px solid var.$black;
				width: 100%;
				position: absolute;

				height: 100%;
				top: 0.5rem;
				left: 0.5rem;
				transition: 0.2s;
			}

			&:hover::after {
				top: 0;
				left: 0;
			}
		}

		& div {
			padding-top: 1.5rem;
			font-family: var.$fedora;
		}
	}

	& .read-the-docs > a {
		color: var.$cyan;
	}
}

@keyframes spin {
	from {
		transform: rotate(0deg);
	}

	to {
		transform: rotate(360deg);
	}
}
