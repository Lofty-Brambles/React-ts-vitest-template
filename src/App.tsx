import React, { useState } from "react";
import reactLogo from "./assets/react.svg";
import "./App.scss";

function App() {
	const [count, setCount] = useState<number>(0);

	return (
		<div className="App">
			<div>
				<img src={reactLogo} className="react" alt="React logo" />
			</div>
			<h1>
				<span>[ âœ¨ ]</span> A Vite template App! <span>[ âœ¨ ]</span>
			</h1>
			<ul>
				<li>ðŸš€ | React with Vite!</li>
				<li>ðŸŽ‰ | Typescript support!</li>
				<li>ðŸ§ª | Vitest for testing!</li>
				<li>ðŸ¥½ | Modular SCSS setup!</li>
				<li>ðŸ¦‹ | ESLint-Prettier with husky and lint-staged!</li>
				<li>ðŸ—º | React-router-dom@v6 and React Icons!</li>
			</ul>
			<div className="card">
				<button type="button" onClick={() => setCount(c => c + 1)}>
					âŒš | Click to brr- {count}
				</button>
				<div>
					<p>
						Edit <code>src/App.tsx</code> and save to test HMR.
					</p>
					<p>
						Style components with <code>.scss</code> component
						files.
					</p>
				</div>
			</div>
			<p className="read-the-docs">
				<a href="https://reactjs.org">React Docs</a> |{" "}
				<a href="https://vitejs.dev/guide/">Vite Guide</a>
			</p>
		</div>
	);
}

export default App;
