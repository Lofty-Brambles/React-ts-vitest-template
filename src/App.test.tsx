/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("Main App Component", () => {
	test("rendering", () => {
		const { container } = render(<App />);
		expect(container).toMatchSnapshot();
	});

	test("tracks click counts once", async () => {
		render(<App />);
		const btn = screen.getByRole("button");

		await userEvent.click(btn);
		expect(btn.textContent).toMatch(/1/);
	});

	test("tracks click count for multiple times", async () => {
		render(<App />);
		const btn = screen.getByRole("button");

		await userEvent.tripleClick(btn);
		expect(btn.textContent).toMatch(/3/);
	});

	test("doesn't reflect changes when clicked elsewhere", () => {
		const { container } = render(<App />);
		const list = screen.getByRole("list");

		userEvent.click(list);

		expect(container).toMatchSnapshot();
	});
});
