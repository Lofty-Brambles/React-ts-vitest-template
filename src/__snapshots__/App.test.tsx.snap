// Vitest Snapshot v1

exports[`Main App Component > doesn't reflect changes when clicked elsewhere 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <img
        alt="React logo"
        class="react"
        src="/src/assets/react.svg"
      />
    </div>
    <h1>
      <span>
        [ âœ¨ ]
      </span>
       A Vite template App! 
      <span>
        [ âœ¨ ]
      </span>
    </h1>
    <ul>
      <li>
        ðŸš€ | React with Vite!
      </li>
      <li>
        ðŸŽ‰ | Typescript support!
      </li>
      <li>
        ðŸ§ª | Vitest for testing!
      </li>
      <li>
        ðŸ¥½ | Modular SCSS setup!
      </li>
      <li>
        ðŸ¦‹ | ESLint-Prettier with husky and lint-staged!
      </li>
      <li>
        ðŸ—º | React-router-dom@v6 and React Icons!
      </li>
    </ul>
    <div
      class="card"
    >
      <button
        type="button"
      >
        âŒš | Click to brr- 
        0
      </button>
      <div>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR.
        </p>
        <p>
          Style components with 
          <code>
            .scss
          </code>
           component files.
        </p>
      </div>
    </div>
    <p
      class="read-the-docs"
    >
      <a
        href="https://reactjs.org"
      >
        React Docs
      </a>
       |
       
      <a
        href="https://vitejs.dev/guide/"
      >
        Vite Guide
      </a>
    </p>
  </div>
</div>
`;

exports[`Main App Component > rendering 1`] = `
<div>
  <div
    class="App"
  >
    <div>
      <img
        alt="React logo"
        class="react"
        src="/src/assets/react.svg"
      />
    </div>
    <h1>
      <span>
        [ âœ¨ ]
      </span>
       A Vite template App! 
      <span>
        [ âœ¨ ]
      </span>
    </h1>
    <ul>
      <li>
        ðŸš€ | React with Vite!
      </li>
      <li>
        ðŸŽ‰ | Typescript support!
      </li>
      <li>
        ðŸ§ª | Vitest for testing!
      </li>
      <li>
        ðŸ¥½ | Modular SCSS setup!
      </li>
      <li>
        ðŸ¦‹ | ESLint-Prettier with husky and lint-staged!
      </li>
      <li>
        ðŸ—º | React-router-dom@v6 and React Icons!
      </li>
    </ul>
    <div
      class="card"
    >
      <button
        type="button"
      >
        âŒš | Click to brr- 
        0
      </button>
      <div>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR.
        </p>
        <p>
          Style components with 
          <code>
            .scss
          </code>
           component files.
        </p>
      </div>
    </div>
    <p
      class="read-the-docs"
    >
      <a
        href="https://reactjs.org"
      >
        React Docs
      </a>
       |
       
      <a
        href="https://vitejs.dev/guide/"
      >
        Vite Guide
      </a>
    </p>
  </div>
</div>
`;
